File: GestorMaestro/GestorMaestro.Web/Modules/Escuela/Sexo/SexoColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GestorMaestro.Escuela.Columns;

[ColumnsScript("Escuela.Sexo")]
[BasedOnRow(typeof(SexoRow), CheckNames = true)]
public class SexoColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int IdSexo { get; set; }
    [EditLink]
    public string Descripcion { get; set; }
}

File: GestorMaestro/GestorMaestro.Web/Modules/ServerTypes/Escuela/SexoColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SexoRow } from "./SexoRow";

export interface SexoColumns {
    IdSexo: Column<SexoRow>;
    Descripcion: Column<SexoRow>;
}

export class SexoColumns extends ColumnsBase<SexoRow> {
    static readonly columnsKey = 'Escuela.Sexo';
    static readonly Fields = fieldsProxy<SexoColumns>();
}